docs/tutorial_1/1_beam_introduction.ipynb

```diff
@@ -1,876 +1,1681 @@
+    "\n",
+    "Our reference `MovieView` dataclass model is very simple, and will represent the atomic **data object** included in the `netflix-prize` `PCollection`. \n",
+    "\n",
+    "To simplify objects instantiation, the `date_stamp` init attribute (i.e. [`InitVar`](https://docs.python.org/3.8/library/dataclasses.html#init-only-variables)), will be later converted with proper data type and format in the `date` field."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dataclasses import dataclass, InitVar\n",
+    "from datetime import datetime\n",
+    "\n",
+    "@dataclass\n",
+    "class MovieView:\n",
+    "    user_id: int\n",
+    "    movie_id: int\n",
+    "    rating: int\n",
+    "    date_stamp: InitVar[str]\n",
+    "    date: datetime.date = None\n",
+    "    \n",
+    "    def __post_init__(self, date_stamp):\n",
+    "        if self.date is None and date_stamp is not None:\n",
+    "            self.date = datetime.strptime(date_stamp, '%Y-%m-%d')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Similarly, the `MovieTitle` data class will encapsulate the `metadata` information extracted per single movie (see `movie_titles.txt` data file:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@dataclass\n",
+    "class MovieTitle:\n",
+    "    movie_id: int\n",
+    "    year: int\n",
+    "    name: str"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "##### Python Type Safety and Custom objects\n",
+    "\n",
+    "The Apache Beam SDK for Python uses type hints during pipeline construction and runtime to try to emulate the correctness guarantees achieved by true static typing. Additionally, using type hints lays some groundwork that allows the backend service to perform efficient type deduction and registration of `Coder` objects.\n",
+    "\n",
+    "Since in this tutorial we want to handle our own custom **data object**, we also need to implement our custom (_yet simple_) `beam.coders.Coder` to allow for a correct object _serialisation_.\n",
+    "\n",
+    "**Note**: This step is particularly crucial during the execution (_in parallel_) of the Pipeline, to avoid `apache_beam` to fallback to the default `Serializer` (i.e. `Coder`) based in `pickle`.\n",
+    "More information on **Python Type Safety** in Apache Beam pipelines is available in the official [documentation](https://beam.apache.org/documentation/sdks/python-type-safety/)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class MovieViewCoder(beam.coders.Coder):\n",
```

Comment:
Could you please add
#@title ....
and hide the code by default (it migth be too much information for the user)